src/data/sigma/order.lean

```diff
@@ -110,6 +110,14 @@ instance [Π i, partial_order (α i)] : partial_order (Σ i, α i) :=
   end,
   .. sigma.preorder }
 
+instance [Π i, preorder (α i)] [Π i, densely_ordered (α i)] : densely_ordered (Σ i, α i) :=
+⟨begin
+  simp_rw sigma.lt_def,
+  rintro ⟨i, a⟩ ⟨j, b⟩ ⟨(rfl : i = j), h⟩,
+  obtain ⟨c, ha, hb⟩ := exists_between h,
+  exact ⟨⟨i, c⟩, ⟨rfl, ha⟩, rfl, hb⟩,
```

Comment:
I don't see the advantage of rewriting by `lt_def`; the proof could be
```suggestion
  rintro ⟨i, a⟩ ⟨_, _⟩ ⟨_, _, b, h⟩,
  obtain ⟨c, ha, hb⟩ := exists_between h,
  exact ⟨⟨i, c⟩, lt.fiber i a c ha, lt.fiber i c b hb⟩,
```
without the rewrite.